### PR TITLE
libgsm 1.0.21

### PR DESCRIPTION
**Pull request by Jerry1144**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

Upstream confirmed the file was updated shortly after release. The header file was changed so we need to build a new bottle. Follows #105038

### CODE REVIEW

**Comment by carlocab** [comment]
Thanks for checking. How did you confirm with upstream?

**Comment by carlocab** [comment]
Looks like there's a race condition in the `Makefile`. Let's try building before installing.

File: Formula/libgsm.rb [hunk 1]
```diff
@@ -2,8 +2,9 @@ class Libgsm < Formula
   desc "Lossy speech compression library"
   homepage "http://www.quut.com/gsm/"
   url "http://www.quut.com/gsm/gsm-1.0.20.tar.gz"
```
Comment (cho-m) [inline Formula/libgsm.rb:4]: How about just updating to **1.0.21**?

**Comment by Jerry1144** [comment]
I wrote an email to the address on the homepage. The author (presumably) says he'd forgot to update the patch-level info in the headers in the initial release. His on-disk file matches the new checksum we get.

> -----Original Message-----
> From: Jutta Degener <jutta@panix.com>
> To: [snip]
> Sent: Fri, Jul 29, 2022 8:38 pm
> Subject: Re: About download of libgsm patch level 20
>
> > I=E2=80=99m writing to ask whether=C2=A0the file =E2=80=98gsm-1.0.20.tar.gz=
> > =E2=80=99=C2=A0from [quut.com](http://quut.com/), the source code of libgsm,=C2=A0has been chan=
> > ged sometime in the past month. Since the site was not using https, I figur=
> > ed the only way to confirm its authenticity=C2=A0was to ask.=C2=A0
> 
> Fair point.  Yes, I did have to update the file a short time after its
> initial publication; I think I may have forgotten to update the patchlevel
> in the gsm.h header file -- so that _would_ affect the binary, although
> it would not affect the behavior.
> 
> Nothing has changed in the behavior for a while; this is now all just
> about mutual battling build systems and, you know, flags.
> 
> And yes, the implied request to move to https is noticed.  It appears
> to also involve a change of hosting altogether, so ... it's complicated.
> 
> Sorry for the noise!
> 
> The sha256sum of the file currently on disk is
> 
> 6315c38518b81e899c3fc2ed463cc623af29c5c231a48c137b24302234ae90bf
> 
> which matches what you got.
> 
> Jutta

Re: build failure. I'm on High Sierra 10.13.6 using CLT 10.14 and the build passes on my machine in 15 seconds. Sorry for the uncaught troubles.

**Comment by Porkepix** [comment]
`1.0.21` is available.

**Comment by carlocab** [comment]
> `1.0.21` is available.

Yes, see https://github.com/Homebrew/homebrew-core/pull/106973#discussion_r934015916

**Comment by Porkepix** [comment]
Damn, missed that one. My bad.

**Comment by Jerry1144** [comment]
Whoops, I never realized a patch was released right after me asking him.

In this new release, the patch level info in `makefile` is still pl-20. This does not affect the binaries in any way (the version variable in the headers is correct) and `brew` doesn't care about that either. Maybe I should not fix what is not broken

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**